enhancements/api-review/azure_user_defined_tags.md

```diff
@@ -0,0 +1,317 @@
+                        for user configuration.
+                        type: array
+                        maxItems: 10
+                        items:
+                          description: AzureResourceTag is a tag to apply to Azure resources created for the cluster.
+                          type: object
+                          required:
+                            - key
+                            - value
+                          properties:
+                            key:
+                              description: key is the key of the tag
+                              type: string
+                              maxLength: 128
+                              minLength: 1
+                              pattern: ^[a-zA-Z][0-9A-Za-z_.=+-@]+$
+                            value:
+                              description: value is the value of the tag.
+                              type: string
+                              maxLength: 256
+                              minLength: 1
+                              pattern: ^[0-9A-Za-z_.=+-@]+$
+```
+
+### Implementation Details/Notes/Constraints [optional]
+Add a new field `resourceTags` to `.status.platformStatus.azure` of the 
+`infrastructure.config.openshift.io` type. Tags included in the `resourceTags` field 
+will be applied to new resources created for the cluster by the in-cluster operators.
+
+The `resourceTags` field in `status.platformStatus.azure` of `infrastructure.config.openshift.io`
+will be populated by the installer using the entries from `platform.azure.userTags` field of `install-config`.
+
+`status.platformStatus.azure` of `infrastructure.config.openshift.io` is immutable.
+
+All operators that create Azure resources will apply these tags to all Azure 
+resources they create. List of in-cluster operators managing cloud resources
+could vary across platform types, example for AWS there are additional operators
+like aws-ebs-csi-driver-operator, aws-efs-csi-driver-operator to manage specific
+resources.
+
+| Operator | Resources created by the operator |
+| -------- | ----------------------------- |
+| cloud-network-config-controller | Private IP address |
+| cluster-image-registry-operator | Storage Account |
+| cluster-ingress-operator | Load Balancer, DNS records |
+| cloud-credential-operator | IAM roles and policies |
+| machine-api-provider-azure | Application Security Group, Availability Set, Group, Load Balancer, Public IP Address, Route, Network Security Group, Virtual Machine Extension, Virtual Interface, Virtual Machine, Virtual Network. |
+
+Below list of terraform Azure APIs to create resources should be updated to add user
+defined tags and as well the openshift default tag in the installer component.
+`azurerm_resource_group, azurerm_image, azurerm_lb, azurerm_network_security_group, azurerm_storage_account, azurerm_user_assigned_identity, azurerm_virtual_network, azurerm_linux_virtual_machine, azurerm_network_interface, azurerm_dns_cname_record`
```

Comment:
I'm not sure I understand the significance of this list. I'm reading this as: this is a list of resources created by the installer that don't have tags and tags will be added to them. Is that correct?

Reply:
Yes, those are the terraform APIs for respective azure resources which needs the change to apply the tags.